docs/src/architecture/08_concepts/hermes_runtime_engine/hre_structure.md

```diff
@@ -0,0 +1,163 @@
+# HRE structure
+
+[Hermes Runtime Extension (*HRE*)][*HRE*] - stands as logically separate module (like a library) of the [*Hermes engine*]
+and provides an additional functionality to the Hermes runtime, therefore to [*Hermes application*].
+[WIT] files represent a source of truth of the [*Hermes events*] and *HRE api* definitions for a specific [*HRE*],
+and describe a standardized communication interface between [*Hermes application*]
+and [*Hermes engine's*][*Hermes engine*] runtime itself.
+
+Each [*HRE*] implementation take place inside `hermes/bin/srs/runtime_extensions` directory,
+for both Hermes related and WASI specific.
+
+Here is an example of the [WIT] files for the [hermes-cron] [*HRE*]:
+
+`world.wit`
+
+```wit
+package hermes:cron;
```

Comment:
Include the contents of the file directly, rather than copy/pasting.
Otherwise if we change the cron .wit files the docs will be immediately out of date.
See the `docs/src/appendix/important/license.md` and `docs/src/api/index.md` files as an example how to do this.
You will also need to include the `.wit` files in the `wasi-hermes-docs` artifact so they are visible to the docs build.

Using the contents of the the `.wit` files directly will also tell us if we break something.  For example rename a file that would make the docs no longer accurate.

Reply:
Alternatively do not refer to a real extension at all, but make an "example" extension which shows off the features you want to describe.  The example extension then just lives in the documentation as a way of explaining how it hangs together, there are no real `.wit` files for it.

Reply:
Got it, yea, make sense

Reply:
fixed, but cannot test because krokie is down 😊